alf/algorithms/distributed_off_policy_algorithm.py

```diff
@@ -491,6 +491,12 @@ def _create_data_receiver_subprocess(self):
         exp = alf.utils.common.prune_exp_replay_state(
             exp, self._use_rollout_state, self.rollout_state_spec,
             self.train_state_spec)
+        # we do not save env_info to replay buffer for the following reasons:
+        # 1) avoid env_info mismatch and allow the distributed unroller to have
+        # a customized env_info for tb summarization, 2) reduce communication overhead,
+
+        # 3) reduce memory usage for the replay buffer
+        exp = alf.utils.common.prune_exp_replay_env_info(exp)
```

Comment:
There is no need to call it here? This exp is obtained from the env itself, so no extra env_info fields exist here.

Reply:
Now we remove env_info completely before saving into replay buffer, not only the mis-match part.

Reply:
> Now we remove env_info completely before saving into replay buffer, not only the mis-match part.

Not sure about this, but did we preserve env_info in the replay buffer for a general usage scenario?

Reply:
Now switched to prune, so only the additional part of env_info will be removed.